RecoTracker/MkFitCore/src/MkFinder.cc

```diff
@@ -2443,6 +2531,425 @@ namespace mkfit {
+        std::cout << "   updated track parameters x=" << m_Par[i2].constAt(i, 0, 0)
+                  << " y=" << m_Par[i2].constAt(i, 1, 0) << " z=" << m_Par[i2].constAt(i, 2, 0) << std::endl;
+        std::cout << "outChi2 " << outChi2[i] << std::endl;
+        std::cout << norm.At(i, 0, 0) << " " << norm.At(i, 1, 0) << " " << norm.At(i, 2, 0) << " "
+                  << "NORM" << std::endl;
+        std::cout << dir.At(i, 0, 0) << " " << dir.At(i, 1, 0) << " " << dir.At(i, 2, 0) << " "
+                  << "DIR" << std::endl;
+        std::cout << pnt.At(i, 0, 0) << " " << pnt.At(i, 1, 0) << " " << pnt.At(i, 2, 0) << " "
+                  << "PNT" << std::endl;
+      }
+#endif
+
+      kalmanPropagateAndUpdateAndChi2Plane(m_Err[i1],
+                                           m_Par[i1],
+                                           m_Chg,
+                                           m_msErr,
+                                           m_msPar,
+                                           norm,
+                                           dir,
+                                           pnt,
+                                           m_Err[i2],
+                                           m_Par[i2],
+                                           m_FailFlag,
+                                           outChi2,
+                                           N_proc,
+                                           *refit_flags,
+                                           true,
+                                           &no_mat_effs);
+
+#ifdef DEBUG_FIT_BKW
+      std::cout << " i1 " << i1 << " iP " << iP << " iC " << iC << std::endl;
+
+      std::cout << "++++++++++++++++++++++++++\n" << std::endl;
+      for (int i = 0; i < N_proc; ++i)  //loop over tracks in group
+      {
+        std::cout << "right after propagation at hit " << h << " index NP " << i + 1 << "/" << N_proc << std::endl;
+        std::cout << "update parameters" << std::endl;
+        std::cout << "propagated track parameters x=" << m_Par[i1].constAt(i, 0, 0)
+                  << " y=" << m_Par[i1].constAt(i, 1, 0) << " z=" << m_Par[i1].constAt(i, 2, 0) << std::endl;
+        std::cout << "               hit position x=" << m_msPar.constAt(i, 0, 0) << " y=" << m_msPar.constAt(i, 1, 0)
+                  << " z=" << m_msPar.constAt(i, 2, 0) << std::endl;
+        std::cout << "   updated track parameters x=" << m_Par[i2].constAt(i, 0, 0)
+                  << " y=" << m_Par[i2].constAt(i, 1, 0) << " z=" << m_Par[i2].constAt(i, 2, 0) << std::endl;
+        std::cout << "outChi2 " << outChi2[i] << std::endl;
+      }
+#endif
+      std::swap(i1, i2);
+
+      // update chi2
+      m_Chi2.add(outChi2);
+
+    }  //end of loop over n hits
+  }  //end of fit func
+
+  //------------------------------------------------------------------------------
+
   void MkFinder::bkFitPropTracksToPCA(const int N_proc) {
     propagateTracksToPCAZ(N_proc, m_prop_config->pca_prop_pflags);
   }
```

Comment:
is it practical to start splitting contributions to more than one file?
If we add smoothing and outlier rejection , it's probably another 1K lines if not more